consensus/babe/babe.go

```diff
@@ -0,0 +1,76 @@
+package babe
+
+import (
+	scale "github.com/ChainSafe/gossamer/codec"
+	"github.com/ChainSafe/gossamer/runtime"
+)
+
+// TODO: change to Schnorrkel keys
+type VrfPublicKey [32]byte
+type VrfPrivateKey [64]byte
+
+// Babe
+type BabeSession struct {
+	vrfPublicKey  VrfPublicKey
+	vrfPrivateKey VrfPrivateKey
+	rt            *runtime.Runtime
+
+	// currentEpoch uint64
+	// currentSlot  uint64
+
+	// TODO: TransactionQueue
+}
+
+// NewBabeSession returns a new Babe session using the provided VRF keys and runtime
+func NewBabeSession(pubkey VrfPublicKey, privkey VrfPrivateKey, rt *runtime.Runtime) *BabeSession {
+	return &BabeSession{
+		vrfPublicKey:  pubkey,
+		vrfPrivateKey: privkey,
+		rt:            rt,
+	}
+}
+
+// BabeConfiguration contains the starting data needed for Babe
+type BabeConfiguration struct {
+	SlotDuration         uint64
+	C1                   uint64 // (1-(c1/c2)) is the probability of a slot being empty
+	C2                   uint64
+	MedianRequiredBlocks uint64
+}
+
+// gets the startup data for Babe from the runtime
+func (b *BabeSession) startupData() (*BabeConfiguration, error) {
+	ret, err := b.rt.Exec("BabeApi_startup_data", 1, 0)
```

Comment:
Lets change this to be a separate function, probably makes sense to group all these calls into a new file. 

We also need to introduce a mutex for the runtime, not required for this PR but might be safer to put it in place now.